common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -1294,6 +1294,173 @@ public boolean toByte(IntWrapper intWrapper) {
     return false;
   }
 
+  /**
+   * Parses this UTF8String(trimmed if needed) to long.
+   *
+   * This method is almost similar to `toLong` defined above. It is used for parsing the UTF8String
+   * when ANSI mode is enabled.
+   * @return If string contains valid numeric value then it returns the long value otherwise a
+   * NumberFormatException  is thrown.
+   */
+
+  public long toLongExact() {
```

Comment:
shall we call this method in `toLong` with try-catch to save duplicated code?

Reply:
Oh. I misunderstood. I will update the code in a bit.

Reply:
@cloud-fan I have updated the code.

Reply:
@cloud-fan I have updated the code.